packages/astro/src/@types/astro.ts

```diff
@@ -1096,12 +1087,111 @@ export interface AstroAdapter {
 
 type Body = string;
 
-export interface APIContext {
+// Shared types between `Astro` global and API context object
+interface AstroSharedContext<Props extends Record<string, any> = Record<string, any>> {
+	/**
+	 * The address (usually IP address) of the user. Used with SSR only.
+	 */
+	clientAddress: string;
+	/**
+	 * Utility for getting and setting cookies values.
+	 */
 	cookies: AstroCookies;
-	params: Params;
+	/**
+	 * Information about the current request. This is a standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) object
+	 */
 	request: Request;
+	/**
+	 * A full URL object of the request URL.
+	 */
+	url: URL;
+	/**
+	 * Parameters passed to a dynamic page generated.
+	 */
+	params: Params;
+	/**
+	 * List of props passed.
+	 */
+	props: Props;
+	/**
+	 * Redirect to another page (**SSR Only**)
+	 */
+	redirect(path: string, status?: number): Response;
+}
+
+export interface APIContext<Props extends Record<string, any> = Record<string, any>>
+	extends AstroSharedContext<Props> {
+	site: URL | undefined;
+	generator: string;
+	/**
+	 * A full URL object of the request URL.
+	 * Equivalent to: `new URL(request.url)`
+	 */
+	url: AstroSharedContext['url'];
+	/**
+	 * Parameters passed to a dynamic page generated using `getStaticPaths`.
```

Comment:
Is this `APIContext` type also applicable in SSR? If so do we need a more generic description?

```suggestion
	 * Parameters matching the page’s dynamic route pattern.
	 * In static builds, this will be the `params` generated by `getStaticPaths`.
	 * In SSR builds, this can be any path segments matching the dynamic route pattern.
```